File: ddz_admin/app/models/bankrupt_save_templates.rb
class BankruptSaveTemplates
   include Mongoid::Document
   field :count, type: Integer
   field :threshold, type: Integer
   field :save_detail, type: Hash, default: ->{ {:_placeholder => 0} }
   include Mongoid::Timestamps
   # field :createdAt, type: Date, default: ->{ Time.now }
   # field :updatedAt, type: Date, default: ->{ Time.now }

    def save_detail_hash
        self.save_detail.to_json
    end

    def save_detail_hash=(value)
        self.save_detail = JSON.parse(value)
    end
end

File: login-server/app/domain/BankruptSaveTemplates.js
/**
 * Created by jeffcao on 15/3/9.
 */
var mongoose = require('mongoose-q')();
var Schema = mongoose.Schema;
var crypto = require('crypto');
var uuid = require('node-uuid');
var DomainUtils = require("./domainUtils");

/**
 * 破产补助模板
 * @type {Mongoose.Schema}
 */
var BankruptSaveTemplateSchema = new mongoose.Schema({
  count: Number,
  threshold: Number,
  save_detail: {type: Schema.Types.Mixed, default: {_placeholder: 0}},       // 补助定义 (自定义配置)
  created_at: {type: Date, default: Date.now},
  updated_at: {type: Date, default: Date.now}
}, {
  collection: 'bankrupt_save_templates'
});


var __toParams = function (model, opts) {
  var transObj = {
    count: model.login_days,
    threshold: model.threshold,
    save_detail: model.save_detail
  };

  transObj = DomainUtils.adjustAttributes(transObj, opts);

  return transObj;
};

BankruptSaveTemplateSchema.statics.toParams = __toParams;

BankruptSaveTemplateSchema.methods.toParams = function (opts) {
  return __toParams(this, opts);
};


var BankruptSaveTemplate = mongoose.model('BankruptSaveTemplate', BankruptSaveTemplateSchema);

module.exports = BankruptSaveTemplate;